tests/component/_task_modules/channels/test_ci_channel.py

```diff
@@ -0,0 +1,385 @@
+import pytest
+
+from nidaqmx import Task
+from nidaqmx._task_modules.channels import CIChannel
+from nidaqmx.constants import (
+    AngleUnits,
+    CountDirection,
+    CounterFrequencyMethod,
+    Edge,
+    EncoderType,
+    EncoderZIndexPhase,
+    UsageTypeCI,
+)
+from nidaqmx.system import Device
+
+
+# Note: Tests for other channel types will be less complete given that the underlying Python
+# implementation is code-generated and the underlying C API implementation is well unit-tested.
+@pytest.mark.parametrize(
+    "decoding_type, zidx_enable, zidx_val, zidx_phase, units, pulses_per_rev, initial_angle, custom_scale_name",
+    [
+        (
+            EncoderType.TWO_PULSE_COUNTING,
+            True,
+            30.0,
+            EncoderZIndexPhase.AHIGH_BHIGH,
+            AngleUnits.DEGREES,
+            24,
+            15.0,
+            "",
+        ),
+        (
+            EncoderType.X_2,
+            False,
+            0.0,
+            EncoderZIndexPhase.AHIGH_BLOW,
+            AngleUnits.RADIANS,
+            12,
+            0.0,
+            "",
+        ),
+        (
+            EncoderType.X_4,
+            False,
+            0.0,
+            EncoderZIndexPhase.AHIGH_BLOW,
+            AngleUnits.FROM_CUSTOM_SCALE,
+            8,
+            0.0,
+            "degrees_scale",
+        ),
+    ],
+)
+def test___task___add_ci_ang_encoder_chan___sets_channel_attributes(
+    task: Task,
+    sim_6363_device: Device,
+    decoding_type: EncoderType,
+    zidx_enable: bool,
+    zidx_val: float,
+    zidx_phase: EncoderZIndexPhase,
+    units: AngleUnits,
+    pulses_per_rev: int,
+    initial_angle: float,
+    custom_scale_name: str,
+):
```

Comment:
FYI, you might as well specify a return type of `-> None`. Otherwise, type checkers assume the function returns `typing.Any`.